wordpress-mobile/WordPress-iOS
# Adds shared git hooks

This introduces a few git hooks to help with development:

- pre-commit: runs rake lint and prevents commits with lint errors.
- post-checkout/post-merge: checks for dependency changes and prompts you to
  run rake dependencies. The most common case would be that Podfile.lock has
  changed and you need to run pod install.

I decided to leave those as opt-in for now, but we could make them mandatory by running `githooks` on `dependencies`.

Also, for the dependency checks, I decided to just print a message as installing can take a while, and I'd rather let the developer decide. Sometimes you just want to quickly switch to another branch to check something in the source, a git hook shouldn't be getting in the way.

To test:

- Run `rake githooks`
- Add a swift style violation to any swift file. Run `git commit` and make sure the hook prevents it
- Merge a branch with CocoaPods changes, after the merge it should show `CocoaPods dependencies missing or outdated. Run rake dependencies to install them.`

Needs review: @astralbodies
